### PR TITLE
Drop-down values when creating loan account at group level aren't available #358 Resolved

### PR DESCRIPTION
**Pull request by mayank-kgp**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.
gradle

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/api/DataManager.java [hunk 1]
```diff
@@ -253,7 +254,11 @@ public DataManager(BaseApiManager baseApiManager,
         return mBaseApiManager.getLoanApi().getAllLoans();
     }
 
-    public Observable<ResponseBody> getGroupLoansAccountTemplate(int groupId, int productId) {
+//    public Observable<ResponseBody> getGroupLoansAccountTemplate(int groupId, int productId) {
+//        return mBaseApiManager.getLoanApi().getGroupLoansAccountTemplate(groupId, productId);
+//    }
```
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/api/DataManager.java]: Please don't submit commented out code in your PR.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java [hunk 2]
```diff
@@ -300,49 +290,27 @@ private void inflateLoanPurposeSpinner() {
         mGroupLoanAccountPresenter.loadGroupLoansAccountTemplate(groupId, productId);
     }
 
-    private void inflateInterestCalculationPeriodSpinner(ResponseBody result) {
+    private void inflateInterestCalculationPeriodSpinner(GroupLoanTemplate groupLoanTemplate) {
 
-        final List<InterestCalculationPeriodType> interestCalculationPeriodType = new
-                ArrayList<>();
+        List<com.mifos.objects.templates.loans.InterestCalculationPeriodType>
```
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java]: Why using class names like this ?

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java [hunk 3]
```diff
@@ -230,39 +236,23 @@ private void inflateLoansProductSpinner() {
...
-                }
-
-            }
-            String stringResult = sb.toString();
-        } catch (Exception e) {
-            Log.e(LOG_TAG, "", e);
+        amortizationType = groupLoanTemplate.getAmortizationTypeOptions();
+
+        for (int i = 0; i < amortizationType.size(); i++) {
+            com.mifos.objects.templates.loans.AmortizationTypeOptions
```
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java]: Bad way of using class names. You should Import these classes.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java [hunk 4]
```diff
@@ -679,6 +596,155 @@ public void onNothingSelected(AdapterView<?> parent) {
...
+                new AdapterView.OnItemSelectedListener() {
+                    @Override
+                    public void onItemSelected(AdapterView<?> adapterView, View view,
+                                               int i, long l) {
+
+                        loanOfficerId = loanOfficerIdHashMap.get
+                                (loanOfficerName.get(i));
+                        Log.d("termFrequencyTypeId" + loanOfficerName.get(i),
+                                String.valueOf(loanTermFrequency));
+                        if (loanTermFrequency != -1) {
```
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java]: No Empty If blocks! This is a bad way of writing code.

**Comment by therajanmaurya** [comment]
@mayank-kgp I appreciated your work done but you have implemented the spinners and other things as the before was implemented. 

Please look  https://github.com/openMF/android-client/tree/master/mifosng-android/src/main/java/com/mifos/mifosxdroid/online/savingsaccount this implementation.

**Comment by mayank-kgp** [comment]
@therajanmaurya  I am changing according to ishan's review. Apart from that, pls come to gitter. We can discuss in detail. :)

**Comment by mayank-kgp** [comment]
@therajanmaurya 
That is nice. I will implement according to savingaccount. 
Thanks for your advice. :)

**Comment by mayank-kgp** [comment]
@therajanmaurya  Please have a look at implementation

**Comment by therajanmaurya** [comment]
@mayank-kgp Good work but can you please make changes that were told by @ishan1604 earlier.
please look above comments.

**Comment by mayank-kgp** [comment]
@therajanmaurya  those are already done when I changed implementation :)
Thanks for checking out implementation. 
@ishan1604  Please do final review as everything look ok.

**Comment by ishan1604** [comment]
@mayank-kgp I don't see the changes anywhere.

File: mifosng-android/src/main/java/com/mifos/api/DataManager.java (hunk 1 above)
Comment (mayank-kgp) [inline mifosng-android/src/main/java/com/mifos/api/DataManager.java]: changed

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java (hunk 3 above)
Comment (mayank-kgp) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java]: changed

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java (hunk 2 above)
Comment (mayank-kgp) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java]: changed

File: mifosng-android/src/main/java/com/mifos/api/services/LoanService.java [hunk 5]
```diff
@@ -85,9 +86,13 @@
     Observable<Loans> createGroupLoansAccount(@Body GroupLoanPayload loansPayload);
 
 
+//    @GET(APIEndPoint.CREATELOANSACCOUNTS + "/template?templateType=group")
+//    Observable<ResponseBody> getGroupLoansAccountTemplate(@Query("groupId") int groupId,
+//                                                          @Query("productId") int productId);
+
```
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/api/services/LoanService.java]: Commented out Code should not be submitted in the PR.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java [hunk 6]
```diff
@@ -103,6 +92,15 @@
     @BindView(R.id.sp_interestcalculationperiod)
     Spinner sp_interestcalculationperiod;
 
+    @BindView(R.id.sp_fund)
+    Spinner sp_fund;
```
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java]: Do not use _ while naming your class member variables. Use Camel Case Only.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountMvpView.java [hunk 7]
```diff
@@ -3,21 +3,22 @@
...
-
 /**
  * Created by Rajan Maurya on 08/06/16.
  */
 public interface GroupLoanAccountMvpView extends MvpView {
 
     void showAllLoans(List<LoanProducts> productLoans);
 
-    void showLoanPurposeSpinner(ResponseBody response);
+  //  void showLoanPurposeSpinner(GroupLoanTemplate groupLoanList);
```
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountMvpView.java]: Commented code

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java [hunk 8]
```diff
@@ -124,5 +139,127 @@ public void onNext(Loans loans) {
...
+        Observable.from(transactionProcessingStrategyOptions)
+                .subscribe(new Action1<TransactionProcessingStrategyOptions>() {
+                    @Override
+                    public void call(TransactionProcessingStrategyOptions
+                                             transactionProcessingStrategyOptions) {
+                        transactionProcessingStrategyNameList.add(
+                                transactionProcessingStrategyOptions.getName());
+                    }
+                });
+       // Log.d("mayanksdf"," "+transactionProcessingStrategyNameList.size());
```
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java]: remove this

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java [hunk 9]
```diff
@@ -135,13 +133,34 @@
...
+    List<String> interestCalculationPeriodType = new ArrayList<>();
+    List<String> transactionProcessingStrategy = new ArrayList<>();
+    List<String> termFrequencyType = new ArrayList<>();
+    List<String> loans = new ArrayList<>();
+    List<String> loanPurposeType = new ArrayList<>();
+    List<String> interestTypeOptions = new ArrayList<>();
+    List<String> loanOfficerOptions = new ArrayList<>();
+    List<String> fundOptions = new ArrayList<>();
+    List<String> mListLoanProductsNames = new ArrayList<>();
+
```
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java]: The lists should be private.

**Comment by ishan1604** [comment]
Please open a new PR with all the changes completed.

**Comment by mayank-kgp** [comment]
@ishan1604  
I am very sorry for this. This was my first PR in which I have written such a long code. I will take care of these points from now. I am opening a new PR. :)

**Pull request closed without merging**